Stripe - Subscriptions Checkout Page - Payment Processing

mathieubouhelier opened the pull request:
Description

As a User, on the BaseApp Profile Page,I would like to subscribe to a plan through the checkout page, In order to complete my subscription payment and access premium features.

 
Acceptance Criteria 
Feature Description

Feature will be done in 2 stories - Payment Method Collection and Payment Processing

Users will reach the Subscription Checkout Page after selecting a subscription plan from the Plans Page.

The page will display:

The selected plan with pricing details.

A payment method selection area, allowing users to:

Choose from previously saved payment methods.

Add a new payment method if none exists.

A billing address input section, used for tax calculations.

A "Place Order" button, to finalize the subscription.

If the user successfully subscribes, a confirmation modal will appear.

AC - Payment Process

Given the user is on the Subscription Checkout Page, when they have saved payment methods, then the default payment method should be preselected, and clicking it should allow selecting a different method.

Given the user has selected a valid payment method, when they click "Place Order", then the system should process the payment via Stripe.

Given the user has previously saved a billing address, when they proceed to checkout, then their saved address should be pre-filled.

Given the payment is successfully processed, when the transaction is complete, then a confirmation modal should appear, displaying the order number and confirmation message.

Order Numbering should follow or use Stripe Invoice number or. ID coming from stripe and not number it on our side

Given the payment fails, when the transaction is declined, then the system should display an error message with retry options. 

Design Link: https://www.figma.com/design/XRD6wSl1m8Kz6XUcAy5CLp/BaseApp---WEB?node-id=7207-4761&t=cmCNSKhQItgRwDB7-4 

Notes 

 

Calculate tax can be covered on a follow-up story

Store everything stripe related on Stripe and we can have follow-up stories to add a caching layer to show Stripe-related information on the UI, but we should avoid storing any Stripe related info on the Admin. 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced a comprehensive Stripe-powered checkout experience, including adding new payment methods, selecting saved cards, and completing subscriptions.
  - Added modals for entering new card details and confirming successful subscription orders.
  - Integrated support for displaying Visa, Mastercard, and generic credit card icons.
  - Enhanced payment method selection with a dropdown and visual feedback.
  - Improved user notifications for errors and successful actions during checkout.

- **Style**
  - Added responsive and accessible layouts for checkout and modal components using Material-UI.

- **Bug Fixes**
  - Improved error handling and user feedback throughout the payment and subscription process.

- **Documentation**
  - Updated and added type definitions for all new components and utilities.

- **Chores**
  - Added Stripe libraries as dependencies and updated package exports for new payment modules.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Test guidance, in addition to the readme guidances:
*In FE-Package*
access to the shell and run this command

```
from django.apps import apps
import swapper
Customer = swapper.load_model("baseapp_payments", "Customer")
customer_entity_model = "users.User"
entity_model = apps.get_model(customer_entity_model)
entity = entity_model.objects.get(profile_id=1)
customer = Customer.objects.create(
  entity=entity,
  remote_customer_id="cus_Rz7mAKnh0TKdAD",)  		
print("Customer created:", customer)
```
Where profile_id is the id of your BA user and remote_customer_id is the ID of the stripe customer

*In FE-Template*
- use the branch "wip-stripe-for-test-purpose"
- update the following value according to your configuration
  ***checkoutProductId={'prod_Rz4zcZaA2WcUaF'}*** 
  ***checkoutCustomerId={'cus_Rz7mAKnh0TKdAD'}*** 
- add "NEXT_PUBLIC_STRIPE_PUBLISHABLE_KEY:"your_value_here"" to the .env file
- run 
``` nvm use```
``` pnpm replace-baseapp-catalogs``` 
``` pnpm clean && pnpm i && pnpm dev``` 
***access to http://localhost:3000/checkout***

mathieubouhelier committed:
1st

rodean1 commented on packages/components/modules/payments/web/CheckoutComponent/Checkout/index.tsx:
```diff
@@ -0,0 +1,238 @@
+    mutate: createSubscription,
+    isPending: isCreatingSubscription,
+    isError: subscriptionError,
+  } = UseCreationSubscription()
```
I would change to `useCreationSubscription`

rodean1 commented on packages/components/modules/payments/web/CheckoutComponent/Checkout/index.tsx:
```diff
@@ -0,0 +1,238 @@
+      },
+    },
+  }
+  console.log('addressOptions', addressOptions)
```
I think these comments can be removed

rodean1 commented on packages/components/modules/payments/web/CheckoutComponent/Checkout/paymentDropDown/index.tsx:
```diff
@@ -0,0 +1,186 @@
+  }, [paymentMethods, setSelectedPaymentMethodId])
+
+  const handleOpenModal = () => {
+    console.log('Opening modal')
```
remove comment

rodean1 commented on packages/components/modules/payments/web/CheckoutComponent/Checkout/paymentDropDown/index.tsx:
```diff
@@ -0,0 +1,186 @@
+  }
+
+  const handleCloseModal = () => {
+    console.log('Closing modal')
```
same here

rodean1 commented on packages/components/modules/payments/web/CheckoutComponent/paymentDropDown/index.tsx:
```diff
@@ -0,0 +1,186 @@
+  }, [paymentMethods, setSelectedPaymentMethodId])
+
+  const handleOpenModal = () => {
+    console.log('Opening modal')
```
remove comment

rodean1 commented on packages/components/modules/payments/web/CheckoutComponent/paymentDropDown/index.tsx:
```diff
@@ -0,0 +1,186 @@
+  }
+
+  const handleCloseModal = () => {
+    console.log('Closing modal')
```
same here

rodean1 commented on packages/components/modules/payments/web/CheckoutComponent/types.ts:
```diff
@@ -0,0 +1,28 @@
+    line2?: string
+    city?: string
+    state?: string
+    postal_code?: string
```
should this be `postalCode` ?

rodean1 commented on packages/components/modules/payments/web/hooks/useStripeHook.tsx:
```diff
@@ -0,0 +1,58 @@
+const useStripeHook = () => {
+  const { sendToast } = useNotification()
+
+  const UseSetupIntent = (customerId?: string) =>
```
`useSetupIntent` ?

rodean1 commented on packages/components/modules/payments/web/hooks/useStripeHook.tsx:
```diff
@@ -0,0 +1,58 @@
+    useMutation({
+      mutationFn: (id: string = customerId || '') => PaymentMethodApi.createSetupIntent(id),
+      onSuccess: (data) => {
+        console.log('Setup intent created successfully:', data)
```
I think this on success should be different than a console log, maybe we can advice on this one

rodean1 commented on packages/components/modules/payments/web/hooks/useStripeHook.tsx:
```diff
@@ -0,0 +1,58 @@
+      mutationFn: (options: CreateSubscriptionOptions) =>
+        PaymentMethodApi.createSubscription(options),
+      onSuccess: (data) => {
+        console.log('Subscription created successfully:', data)
```
this can be a toast

rodean1 commented on packages/components/modules/payments/web/hooks/useStripeHook.tsx:
```diff
@@ -0,0 +1,58 @@
+      enabled: !!customerId,
+    })
+
+  const UseCreationSubscription = () =>
```
`useCreationSubscription` ?

rodean1 commented on packages/components/modules/payments/web/types/index.ts:
```diff
@@ -0,0 +1,60 @@
+      line2?: string
+      city?: string
+      state?: string
+      postal_code?: string
```
`postalCode` ?

anicioalexandre commented on packages/components/modules/payments/stripe.ts:
```diff
@@ -0,0 +1,10 @@
+import { loadStripe } from '@stripe/stripe-js'
+
+const stripePublishableKey = process.env.NEXT_PUBLIC_STRIPE_PUBLISHABLE_KEY
```
We shouldn't put next public env variables into constants, otherwise it wont be inlined during build
https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables#bundling-environment-variables-for-the-browser

anicioalexandre commented on packages/components/modules/payments/stripe.ts:
I think we should move this to `/web/(utils|contants|types).ts` files
This payments implementation is probably going to be unique for web, and for mobile we might need to do a different one. SO I'm afraid we wont reuse anything for now

anicioalexandre commented on packages/components/modules/payments/web/CheckoutComponent/AddCardModal/index.tsx:
```diff
@@ -0,0 +1,150 @@
+  }
+
+  return (
+    <Modal open={open} onClose={onClose}>
```
can we use the BA [Dialog](https://github.com/silverlogic/baseapp-frontend/blob/master/packages/design-system/components/web/dialogs/Dialog/index.tsx) or the [ConfirmDialog](https://github.com/silverlogic/baseapp-frontend/blob/master/packages/design-system/components/web/dialogs/ConfirmDialog/index.tsx)? it comes in with a lot of features

masfiqurtsl commented:
@mathieubouhelier In the mobile view, I'm not sure if we are fine with showing the "Product" and "Place Order" buttons at the top, while the "Add Payment Method" button appears below the "Place Order" button.
<img width="411" alt="Screenshot 2025-05-01 at 3 41 51 AM" src="https://github.com/user-attachments/assets/661c5b0a-2432-4f35-9c85-d6a15778d666" />

masfiqurtsl commented:
When i click on Place order button without adding a payment method the error message only appearing under Name and address file but not under Add Payment method section indicating user need to add a payment method before Placing an oder.  
<img width="1326" alt="Screenshot 2025-05-01 at 3 50 29 AM" src="https://github.com/user-attachments/assets/b4102937-0616-4b5d-9784-da72298e973b" />

masfiqurtsl commented:
@mathieubouhelier The added payment method does not appear instantly for the user after adding it. Instead, an error message shows up under the Name and address fields, and only after a few seconds does the newly added card appear. Can we improve this please. 

Please see this video: https://www.loom.com/share/d85d681276dd44118087cca265933f3e

masfiqurtsl commented:
@mathieubouhelier When i try to add a 2nd Card i am getting an error. 
Please see this video: https://www.loom.com/share/f686277be8c54714ad5386641f047971
<img width="1728" alt="Screenshot 2025-05-03 at 3 11 32 AM" src="https://github.com/user-attachments/assets/de1adc78-9631-4b48-a419-78945e08c850" />

luciobj commented on packages/components/modules/payments/web/CheckoutComponent/AddCardModal/index.tsx:
```diff
@@ -0,0 +1,165 @@
+  }
+  return (
+    <Modal open={open} onClose={onClose}>
+      <Box
```
Maybe move this to a styled component?

luciobj commented on packages/components/modules/payments/web/CheckoutComponent/AddCardModal/index.tsx:
```diff
@@ -0,0 +1,165 @@
+            </Box>
+          </>
+        ) : (
+          <Typography>Loading...</Typography>
```
Maybe use Loading State component, or a circular progress?

luciobj commented on packages/components/modules/payments/web/CheckoutComponent/Checkout/index.tsx:
```diff
@@ -0,0 +1,281 @@
+                </Typography>
+              </Box>
+            </ProductContainer>
+            <Box sx={{ mt: 2 }}>
```
Maybe use gap on the parent component instead of margin here and on the box bellow

luciobj commented on packages/components/modules/payments/web/CheckoutComponent/AddCardModal/index.tsx:
```diff
@@ -0,0 +1,165 @@
+import { Stripe, StripeElements } from '@stripe/stripe-js'
+import { useQueryClient } from '@tanstack/react-query'
+
+interface AddCardModalProps {
```
Move this to a types file

luciobj commented on packages/components/modules/payments/web/CheckoutComponent/ConfirmationSubscriptionModal/index.tsx:
```diff
@@ -0,0 +1,102 @@
+
+import { maskEmail } from '../utils'
+
+interface ConfirmationSubscriptionModalProps {
```
Move this to a types file

masfiqurtsl commented:
This branch is also covering this story: https://silverlogic.atlassian.net/browse/BA-2282

luciobj commented on packages/components/modules/payments/web/CheckoutComponent/ConfirmationSubscriptionModal/index.tsx:
```diff
@@ -0,0 +1,102 @@
+
+  return (
+    <Modal open={open} onClose={onClose} aria-labelledby="subscription-confirmation-title">
+      <Box
```
Add a styled component for this box

luciobj commented on packages/components/modules/payments/web/CheckoutComponent/paymentDropDown/index.tsx:
```diff
@@ -0,0 +1,181 @@
+import { IStripePaymentMethod } from '../../types'
+import AddCardModal from '../AddCardModal'
+
+interface PaymentDropdownProps {
```
Move this to a types file

luciobj commented on packages/components/modules/payments/web/CheckoutComponent/paymentDropDown/index.tsx:
```diff
@@ -0,0 +1,181 @@
+  return (
+    <>
+      {isEmpty ? (
+        <Button
```
Add this button to a styled file

luciobj commented on packages/components/modules/payments/web/hooks/useStripeHook.tsx:
```diff
@@ -0,0 +1,58 @@
+    useMutation({
+      mutationFn: (id: string = customerId || '') => PaymentMethodApi.createSetupIntent(id),
+      onSuccess: () => {
+        console.log('Setup intent created successfully:')
```
Is this log needed?

luciobj commented on packages/components/modules/payments/web/services/stripe.ts:
```diff
@@ -0,0 +1,60 @@
+    allowIncomplete,
+    billingDetails,
+  }: CreateSubscriptionOptions): Promise<ISubscription> => {
+    interface SubscriptionRequestBody {
```
Move this to a types file

luciobj commented on packages/design-system/components/web/icons/CreditCardIcon/index.tsx:
```diff
@@ -0,0 +1,18 @@
+        fillRule="evenodd"
+        clipRule="evenodd"
+        d="M30 6H6V4.875C6 3.83947 6.67157 3 7.5 3H28.5C29.3284 3 30 3.83947 30 4.875V6ZM30 9.75V19.5C30 20.3284 29.3284 21 28.5 21H7.5C6.67157 21 6 20.3284 6 19.5V9.75H30ZM12 15C11.1716 15 10.5 15.6716 10.5 16.5C10.5 17.3284 11.1716 18 12 18H13.5C14.3284 18 15 17.3284 15 16.5C15 15.6716 14.3284 15 13.5 15H12Z"
+        fill="#212B36"
```
This should have the argument "currentColor", instead of a fixed color

masfiqurtsl commented:
@masfiqurtsl All the fixes now working fine. Approving this PR. 
Please see this video: https://www.loom.com/share/9a470ae1a8cb47eeba2e8826c2d897d1

mathieubouhelier committed:
lint

luciobj commented on packages/components/modules/payments/web/services/stripe.ts:
```diff
@@ -0,0 +1,53 @@
+      remote_customer_id: customerId,
+      price_id: priceId,
+    }
+    if (paymentMethodId) {
```
These are not wrong, but they could be simplified by one object options/payload, or even just an inline ternary for the object requestBody, something like this:
  ```
  const requestBody: SubscriptionRequestBody = {
      remote_customer_id: customerId,
      price_id: priceId,
     ...(paymentMethodId & {payment_method_id: paymentMethodId}),
     (...)
    }
```

mathieubouhelier committed:
lint

anicioalexandre commented on packages/components/modules/payments/web/CheckoutComponent/Checkout/index.tsx:
```diff
@@ -0,0 +1,282 @@
+              setIsConfirmCardPaymentProcessing(false)
+              const { paymentIntent } = result
+              setOrderNumber(paymentIntent?.id ?? null)
+              if (result.error) {
```
is there a chance to have `result.error` on the onSuccess callback?

mathieubouhelier replied:
Yes, since it can come from another call inside the onSuccess  =>
`await stripe.confirmCardPayment(clientSecret`

anicioalexandre commented on packages/components/modules/payments/web/CheckoutComponent/Checkout/index.tsx:
```diff
@@ -0,0 +1,282 @@
+                {Number.isFinite(product?.defaultPrice?.unitAmount) && (
+                  <Box sx={{ display: 'flex', justifyContent: 'flex-end', gap: 1 }}>
+                    <Typography variant="body2" fontWeight={700}>
+                      {(product?.defaultPrice?.unitAmount || 0 / 100).toLocaleString('en-US', {
```
I think we could probably extract this into a function, we might need this formatting on other pages

anicioalexandre commented on packages/components/modules/payments/web/CheckoutComponent/index.tsx:
```diff
@@ -0,0 +1,75 @@
+  stripePublishableKey: string
+}
+
+const CheckoutWrapper: FC<CheckoutWrapperProps> = ({
```
lets rename this to CheckoutComponent and move its types to the types.ts file

anicioalexandre commented on packages/components/modules/payments/web/CheckoutComponent/index.tsx:
```diff
@@ -0,0 +1,75 @@
+    createSetupIntent(checkoutCustomerId)
+  }
+
+  return isPending ||
```
could we do a condition instead of a ternary? just because it's easier to read and helps if we ever have another screen to show
```
if (isLoading) return <LoadingState />
return <Elements />
```

priscilladeroode replied:
Can we extract all this checks and make it more readable?

anicioalexandre commented on packages/components/modules/payments/web/CheckoutComponent/paymentDropDown/index.tsx:
```diff
@@ -0,0 +1,153 @@
+
+  return (
+    <>
+      {isEmpty ? (
```
I would move this to a condition too, just for better readability

mathieubouhelier replied:
Can we keep it? Otherwise, we will have to duplicate the <AddCardModal>, I believe

anicioalexandre commented on packages/components/modules/payments/web/CheckoutComponent/paymentDropDown/index.tsx:
```diff
@@ -0,0 +1,153 @@
+            labelId="payment-method-select-label"
+            value={selectedPaymentMethodId}
+            label="Select a Payment Method"
+            onChange={(e) => {
```
lets move this into a function so we separate the logic from the view a bit more

anicioalexandre commented on packages/components/modules/payments/web/types/index.ts:
```diff
@@ -0,0 +1,61 @@
+  clientSecret: string
+}
+
+export interface IStripePaymentMethod {
```
can we avoid the I prefix, please?

priscilladeroode replied:
Don't we have something that can be exported from the stripe library?

anicioalexandre commented on packages/components/modules/payments/web/CheckoutComponent/paymentDropDown/index.tsx:
small typo on the folder name, `PaymentDropdown`

anicioalexandre replied:
@mathieubouhelier can we rename it?

anicioalexandre commented on packages/components/modules/payments/web/services/types.ts:
lets move these types to `packages/components/modules/payments/web/types/stripe.ts`

anicioalexandre commented on packages/components/modules/payments/web/types/index.ts:
lets move these types to `packages/components/modules/payments/web/types/stripe.ts`

anicioalexandre commented on packages/components/modules/payments/web/hooks/useStripeHook.tsx:
```diff
@@ -0,0 +1,55 @@
+
+  const useGetPaymentMethod = (customerId: string) =>
+    useQuery({
+      queryKey: ['useGetPaymentMethod', customerId],
```
It would be cool to have the query keys next to the API service class, so it's easy to reuse alongside the components
```
// on services/stripe.ts
export const STRIPE_API_KEY = {
  default: ['stripe'],
  getPaymentMethod: (customerId: string) => [...STRIPE_API_KEY.default, 'getPaymentMethod', customerId],
  getProduct: (customerId: string) => [...STRIPE_API_KEY.default, 'getProduct', customerId],
  ...
}
```

anicioalexandre commented on packages/components/modules/payments/web/CheckoutComponent/Checkout/index.tsx:
```diff
@@ -0,0 +1,282 @@
+  const { mutate: createSubscription, isPending: isCreatingSubscription } =
+    useCreationSubscription()
+
+  const selectedMethod = useMemo(
```
why do we need this since we already have `selectedPaymentMethodId/setSelectedPaymentMethodId` states?

mathieubouhelier replied:
yes, because we need data from selectedMethod (selectedMethod?.billingDetails?.address?.line1 ...)

anicioalexandre commented on packages/components/modules/payments/web/CheckoutComponent/Checkout/index.tsx:
```diff
@@ -0,0 +1,282 @@
+            const { clientSecret } = data
+            if (stripe && clientSecret) {
+              setIsConfirmCardPaymentProcessing(true)
+              const result = await stripe.confirmCardPayment(clientSecret, {
```
can we extract this into a hook? maybe using react query mutation, so we already have the `isLoading` states

priscilladeroode commented on packages/components/modules/payments/web/CheckoutComponent/AddCardModal/index.tsx:
```diff
@@ -0,0 +1,157 @@
+  const isMobile = useMediaQuery<Theme>((theme) => theme.breakpoints.down('sm'))
+  const isTablet = useMediaQuery<Theme>((theme) => theme.breakpoints.between('sm', 'md'))
+  const [isAddingCardPaymentProcessing, setIsAddingCardPaymentProcessing] = useState(false)
+  console.log('isAddingCardPaymentProcessing', isAddingCardPaymentProcessing)
```
Can you remove?

priscilladeroode commented on packages/components/modules/payments/web/CheckoutComponent/AddCardModal/index.tsx:
```diff
@@ -0,0 +1,157 @@
+
+    const addressElement = elements.getElement(AddressElement)
+    if (!addressElement) {
+      console.error('AddressElement not found')
```
Can you remove?

priscilladeroode commented on packages/components/modules/payments/web/CheckoutComponent/AddCardModal/index.tsx:
```diff
@@ -0,0 +1,157 @@
+
+    const addressValue = await addressElement.getValue()
+    if (!addressValue.complete) {
+      console.error('Address validation error: Incomplete address')
```
Can you remove?

priscilladeroode commented on packages/components/modules/payments/web/CheckoutComponent/AddCardModal/index.tsx:
```diff
@@ -0,0 +1,157 @@
+      })
+
+      if (result.error) {
+        console.error('Error confirming card:', result.error.message)
```
Can you remove?

priscilladeroode commented on packages/components/modules/payments/web/CheckoutComponent/paymentDropDown/index.tsx:
```diff
@@ -0,0 +1,153 @@
+    const cardBrand = brand?.toLowerCase() || ''
+
+    switch (cardBrand) {
+      case 'visa':
```
Create constant for Card Brands

priscilladeroode commented on packages/components/modules/payments/web/CheckoutComponent/Checkout/index.tsx:
```diff
@@ -0,0 +1,282 @@
+                if (paymentMethods?.length > 0) {
+                  setSelectedPaymentMethodId(paymentMethods[0]?.id || '')
+                }
+                queryClient.invalidateQueries({ queryKey: ['useGetPaymentMethod'] })
```
Can you create constants for all the queryKeys you added?

priscilladeroode commented on packages/components/modules/payments/web/CheckoutComponent/Checkout/index.tsx:
```diff
@@ -0,0 +1,282 @@
+                  <Box sx={{ display: 'flex', justifyContent: 'flex-end', gap: 1 }}>
+                    <Typography variant="body2" fontWeight={700}>
+                      {(product?.defaultPrice?.unitAmount || 0 / 100).toLocaleString('en-US', {
+                        style: 'currency',
```
Can we make the currency configurable?

priscilladeroode commented on packages/components/modules/payments/web/CheckoutComponent/paymentDropDown/types.tsx:
```diff
@@ -0,0 +1,15 @@
+import { Stripe, StripeElements } from '@stripe/stripe-js'
```
This file can be a .ts instead of tsx

priscilladeroode commented on packages/components/modules/payments/web/CheckoutComponent/Checkout/index.tsx:
```diff
@@ -0,0 +1,282 @@
+          </Box>
+        </Grid>
+      </Grid>
+      <ConfirmationSubscriptionModal
```
Can we make this ConfirmationSubscriptionModal more "customizable"?
Can you add on the props something like ConfirmationSubscriptionModal = DefaultConfirmationSubscriptionModal so we can allow people to build their own modal if they want?

priscilladeroode commented on packages/components/modules/payments/web/CheckoutComponent/types.ts:
```diff
@@ -0,0 +1,30 @@
+  onPaymentError?: (error: string) => void
+}
+
+export interface BillingDetails {
```
BillingDetailsProps

priscilladeroode commented on packages/components/modules/payments/web/types/index.ts:
```diff
@@ -0,0 +1,61 @@
+    expMonth: number
+    expYear: number
+  }
+  billingDetails?: {
```
Why are we not using the BillingDetailsProps here?

priscilladeroode commented on packages/components/modules/payments/web/CheckoutComponent/types.ts:
```diff
@@ -0,0 +1,30 @@
+export interface BillingDetails {
+  name?: string
+  address?: {
+    line1?: string
```
We are reusing the address type in a lot of places, let's create an interface for address and reuse it on every place you have the address type.

anicioalexandre commented on packages/components/modules/payments/web/CheckoutComponent/Checkout/index.tsx:
```diff
@@ -0,0 +1,282 @@
+    [selectedPaymentMethodId, paymentMethods],
+  )
+
+  useEffect(() => {
```
do we need this effect? 
Does `hasMounted` mean when we have `paymentMethods` fetched? Is there a point for loading the checkout component without having the payment methods fetched? Maybe we can simplify the logic a bit here

mathieubouhelier committed:
wip

mathieubouhelier committed:
wip2

mathieubouhelier committed:
addressed comments

mathieubouhelier committed:
add Dialog

mathieubouhelier committed:
remove useEffect

lokmanliton commented:
Hello, I got some design missmatch here as per design, tested so far on web, will test on mobile and tablet too, 
here are my issues

1. Gap Between menu bar and checkout frame is not same as design.
2. In implementation, we don't have Checkout header as we have in design.
3. Field Names are not same as design, 
   ** like in design, Under payment, we don't have drop-down name for payment method, but in implementation it has field title for payment method that squeezes the drop-down box a bit.
  ** Here in design it says Address, but in implementation it says Billing Address.
  ** In design we don't have name field even, we have Country, Address line 1, Address Line 2, City, ZIP code and state, 
  *** But in implementation, we have full name, Country or region, Address line 1, Address line 2, City, State and ZIP code, but not aligned as per design.
4. Add payment Method also have some UI mismatch as per design.

Please watch this loom for more clarification - https://www.loom.com/share/a2e90c79483f455bb6003a7dc0f142ec

mathieubouhelier committed:
design fixes

mathieubouhelier commented:
> Hello, I got some design missmatch here as per design, tested so far on web, will test on mobile and tablet too, here are my issues
> 
> 1. Gap Between menu bar and checkout frame is not same as design.
addressed
> 2. In implementation, we don't have Checkout header as we have in design.
addressed
> 3. Field Names are not same as design,

>    ** like in design, Under payment, we don't have drop-down name for payment method, but in implementation it has field title for payment method that squeezes the drop-down box a bit.
addressed
>    ** Here in design it says Address, but in implementation it says Billing Address.
addressed
>    ** In design we don't have name field even, we have Country, Address line 1, Address Line 2, City, ZIP code and state,
>    *** But in implementation, we have full name, Country or region, Address line 1, Address line 2, City, State and ZIP code, but not aligned as per design.
> 4. Add payment Method also have some UI mismatch as per design.
> 
> Please watch this loom for more clarification - https://www.loom.com/share/a2e90c79483f455bb6003a7dc0f142ec

Please remember that the address and add payment method are imported from Stripe and cannot be customized.

lokmanliton commented:
> > Hello, I got some design missmatch here as per design, tested so far on web, will test on mobile and tablet too, here are my issues
> > 
> > 1. Gap Between menu bar and checkout frame is not same as design.
> >    addressed
> > 2. In implementation, we don't have Checkout header as we have in design.
> >    addressed
> > 3. Field Names are not same as design,
> 
> > ** like in design, Under payment, we don't have drop-down name for payment method, but in implementation it has field title for payment method that squeezes the drop-down box a bit.
> > addressed
> > ** Here in design it says Address, but in implementation it says Billing Address.
> > addressed
> > ** In design we don't have name field even, we have Country, Address line 1, Address Line 2, City, ZIP code and state,
> > *** But in implementation, we have full name, Country or region, Address line 1, Address line 2, City, State and ZIP code, but not aligned as per design.
> > 4. Add payment Method also have some UI mismatch as per design.
> > Please watch this loom for more clarification - https://www.loom.com/share/a2e90c79483f455bb6003a7dc0f142ec
> 
> Please remember that the address and add payment method are imported from Stripe and cannot be customized.

Thanks for the fixing, 
I am approving the PR with two small design issue, we can fix those later, 
1. Implementation has excess left margin between sidebar menu and checkout frame.
<img width="1787" alt="Screenshot 2025-05-26 at 10 28 22 PM" src="https://github.com/user-attachments/assets/41287fba-316b-47b1-8ae9-bc5f6e06050f" />
2. Padding in both sides should be 32px as per design, but we have 16px in implementation, 
<img width="1787" alt="Screenshot 2025-05-26 at 10 54 43 PM" src="https://github.com/user-attachments/assets/cd06d3aa-fcd8-47e6-8976-9e7f219f692b" />

anicioalexandre reviewed (approved):
ufff! that one was a big one! good job @mathieubouhelier 
could you fix the ci so we can merge it?
I believe we might need to rename this branch and merge it into an epic branch, right @nossila ?

mathieubouhelier committed:
Temp rename to trigger case change

mathieubouhelier committed:
Rename to PaymentDropDown

nossila commented:
> ufff! that one was a big one! good job @mathieubouhelier could you fix the ci so we can merge it? I believe we might need to rename this branch and merge it into an epic branch, right @nossila ?

@anicioalexandre we don't need to rename old branches, renaming it closes the PR and opens it again, which breaks the code review flow... I added a rule for new branches though, you shouldn't be able to create new branches without the correct prefix.

mathieubouhelier committed:
versioning
